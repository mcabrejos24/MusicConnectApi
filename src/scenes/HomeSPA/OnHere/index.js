import './style.scss';
import { useState } from 'react';
import AuthorizerAppleMusic from '../../../components/AuthorizerAppleMusic';
import AuthorizerSpotify from '../../../components/AuthorizerSpotify';
import MusicCard from '../../../components/MusicCard';
import SyncElement from '../../../components/SyncElement';
import BackButton from '../components/BackButton';

export default function OnHere() {
    const [appleReady, setAppleReady] = useState(false);
    const [spotifyReady, setSpotifyReady] = useState(false);

    return (
        <div className="content">
            <BackButton></BackButton>
            <div className="authorization-wrapper">
                {/* Apple Music */}
                <MusicCard service={'apple'} link={'https://music.apple.com/us/browse'} linkName={'Apple'} confirmPlaylist={setAppleReady}>
                    <AuthorizerAppleMusic></AuthorizerAppleMusic>
                </MusicCard>

                {/* Spotify */}
                <MusicCard service={'spotify'} link={'https://www.spotify.com'} linkName={'Spotify'} confirmPlaylist={setSpotifyReady}>
                    <AuthorizerSpotify></AuthorizerSpotify>
                </MusicCard>
            </div>
            <SyncElement spotifyConfirmed={spotifyReady} appleConfirmed={appleReady}></SyncElement>

            <div className="how-it-works">
                <h1 className="how-it-works-title">How to use this sync option</h1>

                <div className="point need">
                    <h2 className="point-title">Cards</h2>
                    <div className="point-description">
                        <p>This one is made for when both you and your peer are together in the same room.
                            <br /><br />
                            Simply, one of you logs into your Apple Music and the other into your Spotify.
                            <br /><br />
                            You both then choose the playlist you would like to sync.
                            <br /><br />
                            If you want to make a new playlist, then enter the name of the new playlist and press the button below it to create it.
                            <br /><br />
                            Then just press the Sync button and your playlists will be synced!
                            <br /><br />
                            The progress bar helps tell you what step you are on.
                        </p>
                    </div>
                </div>
            </div>



        </div>
    );
}
