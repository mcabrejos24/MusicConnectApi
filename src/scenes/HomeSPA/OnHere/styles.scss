.authorization-wrapper {
    display: grid;
    grid-template-rows: 1fr 1fr;
    padding: 100px 0;
    justify-items: center;
    row-gap: 130px;

    @media screen and (min-width: 768px) {
        row-gap: unset;
        grid-template-rows: unset;
        grid-template-columns: 1fr 1fr;

    }

    a {
        display: flex;
        justify-content: center;
        text-align: center;
    }
}
