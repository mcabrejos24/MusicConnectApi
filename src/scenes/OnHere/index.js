import React from 'react';

export default function OnHere() {

    return (
        <div className="App">
            <p>OnHere Page</p>
        </div>
    );
}
