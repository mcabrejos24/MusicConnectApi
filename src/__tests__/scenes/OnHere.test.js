import React from "react";
import { shallow } from "enzyme";
import OnHere from '../../scenes/HomeSPA/OnHere';
import toJson from "enzyme-to-json";

describe("renders OnHere page", () => {
    let page;
    beforeEach(() => {
        page = shallow(<OnHere />);
    });
    it("OnHere loads", () => {
        page;
    });
    it("props class name matches", () => {
        expect(page.props().className).toEqual("content on-here-page");
    });
    it("matches snapshot", () => {
        expect(toJson(page)).toMatchSnapshot();
    });
});
