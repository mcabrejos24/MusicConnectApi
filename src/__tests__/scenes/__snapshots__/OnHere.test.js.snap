// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders OnHere page matches snapshot 1`] = `
<div
  className="content on-here-page"
>
  <BackButton />
  <div
    className="authorization-wrapper"
  >
    <MusicCard
      confirmPlaylist={[Function]}
      link="https://music.apple.com/us/browse"
      linkName="Apple"
      service="apple"
    >
      <AuthorizerAppleMusic />
    </MusicCard>
    <MusicCard
      confirmPlaylist={[Function]}
      link="https://www.spotify.com"
      linkName="Spotify"
      service="spotify"
    >
      <AuthorizerSpotify />
    </MusicCard>
  </div>
  <SyncElement
    appleConfirmed={false}
    spotifyConfirmed={false}
  />
  <div
    className="how-it-works on-here"
  >
    <h1
      className="how-it-works-title"
    >
      How to use this sync option
    </h1>
    <div
      className="point need"
    >
      <h2
        className="point-title"
      >
        Cards
      </h2>
      <div
        className="point-description"
      >
        <p>
          This syncing method is for when both you and your peer are together in the same room.
          <br />
          <br />
          Or if you have access to both services yourself.
          <br />
          <br />
          Simply, one of you logs into your Apple Music account and the other into your Spotify.
          <br />
          <br />
          You both then choose the playlist you would like to sync.
          <br />
          <br />
          If you want to make a new playlist, then enter the name of the new playlist and press the button below it to create it.
          <br />
          <br />
          Note that for Apple Music, you will need to add a song after creating a new playlist on here, or on the service.
          <br />
          <br />
          Then just press the Sync button and your playlists will be synced!
          <br />
          <br />
          The progress bar helps tell you what step you are on.
          <br />
          <br />
        </p>
      </div>
    </div>
  </div>
</div>
`;
